RecoLocalCalo/HcalRecAlgos/src/ZdcSimpleRecAlgo_Run3.cc

```diff
@@ -0,0 +1,547 @@
+    // else return(.4*energy1);
+    return (.4 * energy1);
+  }
+  inline double calcNoiseRPD(const float energy0, const float energy1) {
+    // if(energy0<=0 || energy1<=0)return (energy1);
+    // // check if noise TS is less than half of previous TS
+    // else if( energy0 >= 2*energy1) return( .4*energy1);
+    // else return(.4*energy1);
+    return (.4 * energy1);
+  }
+
+}  // namespace zdchelper
+
+namespace ZdcSimpleRecAlgoImpl {
+  template <class Digi, class RecHit>
+  inline RecHit reco1(const Digi& digi,
+                      const HcalCoder& coder,
+                      const HcalCalibrations& calibs,
+                      const HcalPedestal& effPeds,
+                      const std::vector<unsigned int>& myNoiseTS,
+                      const std::vector<unsigned int>& mySignalTS) {
+    CaloSamples tool;
+    coder.adc2fC(digi, tool);
+    // Reads noiseTS and signalTS from database
+    int ifirst = mySignalTS[0];
+    double ampl = 0;
+    int maxI = -1;
+    double maxA = -1e10;
+    double ta = 0;
+    double lowGEnergy = 0;
+    double energySOIp1 = 0;
+    double ratioSOIp1 = -1.0;
+    double chargeWeightedTime = 0;
+
+    double Allnoise = 0;
+    int noiseslices = 0;
+    int CurrentTS = 0;
+    double noise = 0;
+    int digi_size = digi.size();
+
+    // determining noise
+    for (unsigned int iv = 0; iv < myNoiseTS.size(); ++iv) {
+      CurrentTS = myNoiseTS[iv];
+      int capid = digi[CurrentTS].capid();
+      // float ped = calibs.pedestal(capid);
+      float ped = effPeds.getValue(capid);
+      float width = effPeds.getWidth(capid);
+      if (CurrentTS >= digi_size)
+        continue;
+      float energy1 = zdchelper::subPedestal(tool[CurrentTS], ped, width);
+      float energy0 = 0;
+      if (CurrentTS > 0)
+        energy0 = zdchelper::subPedestal(tool[CurrentTS - 1], ped, width);
+      Allnoise += zdchelper::calcNoise(energy0, energy1);
+      noiseslices++;
+    }
+    if (noiseslices != 0) {
+      noise = (Allnoise) / double(noiseslices);
+    } else {
+      noise = 0;
+    }
+
+    for (unsigned int ivs = 0; ivs < mySignalTS.size(); ++ivs) {
+      CurrentTS = mySignalTS[ivs];
+      if (CurrentTS >= digi_size)
+        continue;
+      float energy1 = -1;
+      int capid = digi[CurrentTS].capid();
+      // float ped = calibs.pedestal(capid);
+      float ped = effPeds.getValue(capid);
+      float width = effPeds.getWidth(capid);
+      float energy0 = std::max(0.0, zdchelper::subPedestal(tool[CurrentTS], ped, width));
+      if (CurrentTS < digi_size - 1)
+        energy1 = std::max(0.0, zdchelper::subPedestal(tool[CurrentTS + 1], ped, width));
+      ratioSOIp1 = (energy0 > 0 && energy1 > 0) ? energy0 / energy1 : -1.0;
+      ta = energy0 - noise;
+      if (ta > 0)
+        ampl += ta;
+
+      if (ta > maxA) {
+        maxA = ta;
+        maxI = CurrentTS;
+      }
+    }
+
+    // LowGainEnergy not used currently
+    lowGEnergy = -99;
```

Comment:
Then please just get rid of this variable (and pass -99 directly if needed, with a comment)

Reply:
lowGEnergy is not passed directly in the ZDCRechit initialization as suggested